### PR TITLE
Fixed the name attribute of file tag when using checkstyle format and no cache

### LINKED ISSUE #6097: The checkstyle format includes the current directory
## Bug report

PHP
```
PHP 7.4.25 (cli) (built: Oct 23 2021 18:05:05) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.25, Copyright (c), by Zend Technologies
```

php-cs-fixer
```
PHP CS Fixer 3.2.1 Mountains by Fabien Potencier and Dariusz Ruminski
```

I use the `--format=checkstyle` option as I am running the output through `cs2pr` in a GH Action. However `cs2pr` fails to add annotation because the file name in the XML out[ut contains the current directory.

I fixed my GH Action to remove it but I consider this a bug in php-cs-fixer.

To reproduce I have used this file

```php
<?php

class Foo
{

}
```

Run the following command 

```
php-cs-fixer fix --dry-run --using-cache=no --format=checkstyle -vvv ./test.php
```

and the output is

```xml
<?xml version="1.0" encoding="UTF-8"?>
<checkstyle>
  <file name="CareLineLive/test.php">
    <error severity="warning" source="PHP-CS-Fixer.method_argument_space" message="Found violation(s) of type: method_argument_space"/>
  </file>
</checkstyle>
```

I believe the `name` attribute should be `name="test.php"` or maybe `name="./test.php"`.

### PR DESCRIPTION
**Pull request by troccoli**
Closes #6097

### CODE REVIEW

**Comment by CLL-GTA** [comment]
Any maintainer wishes to have a look at this?

File: tests/Smoke/StdinTest.php [hunk 1]
```diff
@@ -46,7 +46,7 @@ public function testFixingStdin()
         );
         static::assertSame($expectedError, $stdinResult->getError());
 
-        $path = str_replace('/', \DIRECTORY_SEPARATOR, basename(realpath($cwd)).'/'.$inputFile);
```
Comment (keradus) [inline tests/Smoke/StdinTest.php:49]: isn't this change braking the usage of fixing the relatives and symlink/linked files?

maybe it's worth to add a test cases like those

File: tests/Console/ConfigurationResolverTest.php [hunk 2]
```diff
@@ -1267,7 +1267,7 @@ public function provideDeprecatedFixerConfiguredCases()
     public function provideGetDirectoryCases()
     {
         return [
```
Comment (keradus) [inline tests/Console/ConfigurationResolverTest.php:1269]: can we add a cases like the following, to show that `basedir` is not messing up with single dot?
`php-cs-fixer fix .`
`php-cs-fixer fix src/File.php`

**Comment by keradus** [comment]
pls target master branch instead, 2.19 is no longer maintained

File: tests/Console/ConfigurationResolverTest.php (hunk 2 above)
Comment (CLL-GTA) [reply tests/Console/ConfigurationResolverTest.php:1269]: So I think I can add the following test cases
```
            [null, './my/file', './my/file'],
            [null, 'my/file', 'my/file'],
            ['/my/path2/dir/.php-cs-fixer.cache', './dir2/file', './dir2/file'],
            ['/my/path2/dir/.php-cs-fixer.cache', 'dir2/file', 'dir2/file'],
            ['dir/.php-cs-fixer.cache', './dir3/file', './dir3/file'],
            ['dir/.php-cs-fixer.cache', 'dir3/file', 'dir3/file'],
```

And they will all pass. But I wonder if when the file starts with `./` we should expect the result without, so for example the first two cases would be
```php
            [null, './my/file', 'my/file'],
            [null, 'my/file', 'my/file'],
```

File: tests/Smoke/StdinTest.php (hunk 1 above)
Comment (CLL-GTA) [reply tests/Smoke/StdinTest.php:49]: I'm not sure I understand what you mean. This is a test and the change adjust what we expect from stdout.

**Comment by CLL-GTA** [comment]
@keradus 

Something went wrong with a merge. Please close this PR. I will make a brand new one.

**Pull request closed without merging**